p2p/pex/pex_reactor.go

```diff
@@ -189,6 +189,13 @@ func (r *PEXReactor) logErrAddrBook(err error) {
 	}
 }
 
+func (r *PEXReactor) InitAddPeer(peer Peer) Peer {
+	id := string(peer.ID())
+	r.requestsSent.Delete(id)
```

Comment:
Can't we just add this to AddPeer?

Reply:
And why isn't it enough for this to be done in RemovePeer?

Reply:
It may do concurrently